reports/vamshi99.md

```diff
@@ -0,0 +1,169 @@
+ - 11:
+   - repo: gitmate-2
+   - link: https://gitlab.com/gitmate/open-source/gitmate-2/commit/db76142d3ecb1e6cd9cd10885ae2475cb07c4d3d
+   - details: >
+      Notify merge conflicts in a PR
+ - 12:
+   - repo: gitmate-2
+   - link: https://gitlab.com/gitmate/open-source/gitmate-2/commit/ad5480c9b0a124d98020a4ad3fd3caf20fa1e321
+   - details: >
+      pr_rejector: Support regex in branch names
+ - 13:
+   - repo: IGitt
+   - link: https://gitlab.com/gitmate/open-source/IGitt/commit/94735ff0f1024d6f293db90516087d8e4d2c353d
+   - details: >
+      Organization: Add issues property and filter_issues method
+ - 14:
+   - repo: IGitt
+   - link: https://gitlab.com/gitmate/open-source/IGitt/commit/fff0d568f1f2731c5055995ff19bbc93be842d5f
+   - details: >
+      Repository: Filter issues by label, assignee
+ - 15:
+   - repo: IGitt
+   - link: https://gitlab.com/gitmate/open-source/IGitt/commit/c025311167b07da5e825433a4d4d5e376adfed62
+   - details: >
+      GitLabUser: Fetch user id from username
+ - 16:
+   - repo: mr-rebaser
+   - link: https://gitlab.com/gitmate/open-source/mr-rebaser/commit/fad8305823a1058e93cd2397b5239a882b66c3f9
+   - details: >
+      Add squash pull request command
+ - 17:
+   - repo: mr-rebaser
+   - link: https://gitlab.com/gitmate/open-source/mr-rebaser/commit/3124c07ca5cd46acf985dc25fe22889e5a4744d0
+   - details: >
+      Dockerfile: Install requirements
+ - 18:
+   - repo: mr-rebaser
+   - link: https://gitlab.com/gitmate/open-source/mr-rebaser/commit/68360f3f0fac686907c7eb44c1a70bfa9e276d42
+   - details: >
+      run.py: Update squash command
+ - 19:
+   - repo: gitmate-2
+   - link: https://gitlab.com/gitmate/open-source/gitmate-2/commit/dda98a8ab67d01376d3199bb82260838cebab7b0
+   - details: >
+      rebaser: Add squash command
+---
+
+### Implement Gitmate automations and plugins for coala
+
+#### Work Done
+
+1. New Plugins added in GitMate:
+   1. Auto-assign issue based on commit mention in a Pull Request.
+   2. Notify merge conflicts in a Pull Request.
+   3. Auto review of Fixes vs closes keywords in a Pull request.
+   4. Auto reject pull requests opened on specified branches.
+   5. Squash commits in a Pull Request.
+   6. Request assignment to an issue.
+
+2. Add WIP label and remove pending review label if coala plugin finds defects
+   in a Pull Request.
+3. IGitt enhancements:
+   1. Filter issues by label, assignee in a repository.
+   2. Fetch user id from username in GitLabUser.
+   3. Added issues property and filter_issues method in Organization class.
+   4. Added fix_issues, close_issues, resolve_issues properties in
+      PullRequest.
+
+#### Challenges
+
+I had difficulties with implementing GitLab API in IGitt. GitHub has teams and
+GitLab API doesn't support teams, so I had to re-design assign request plugin
+to work without teams. Similarly, GitLab API doesn't have any webhook for pull
+request review event, so I couldn't implement one of the proposed features.
```

Comment:
which proposed feature wasnt implemented?

Reply:
GitMate labelling PR as per review done by maintainer.

Reply:
So that needs to be added to "Work to be done" below?

Or create a separate section for this?

All projects drop parts of the requirements deemed to difficult.

It would be great if we can be explicit/honest/clear about this, and explain why it was dropped.

Reply:
>GitLab API doesn't have any webhook for pull
request review event
@jayvdb ^